matchms/Scores.py

```diff
@@ -192,12 +198,12 @@ def scores_by_query(self, query: QueriesType) -> numpy.ndarray:
         """
         assert query in self.queries, "Given input not found in queries."
         selected_idx = int(numpy.where(self.queries == query)[0])
-        selected_scores = []
-        for i, score in enumerate(self._scores[:, selected_idx].copy()):
-            if isinstance(score, tuple):
-                selected_scores.append((self.references[i], *score))
-            else:
-                selected_scores.append((self.references[i], score))
+        if sort:
+            references_idx_sorted = self.similarity_function.sort(self._scores[:, selected_idx])
+            selected_scores = list(zip(self.references[references_idx_sorted],
+                                       self._scores[references_idx_sorted, selected_idx].copy()))
+        else:
+            selected_scores = list(zip(self.references, self._scores[:, selected_idx].copy()))
         return selected_scores
```

Comment:
Could be simplified to
```suggestion
            return list(zip(self.references[references_idx_sorted],
                                       self._scores[references_idx_sorted, selected_idx].copy()))
        return list(zip(self.references, self._scores[:, selected_idx].copy()))
```